packages/apollo-server-errors/src/index.ts

```diff
@@ -14,7 +25,7 @@ export class ApolloError extends Error implements GraphQLError {
   readonly source: Source | undefined;
   readonly positions: ReadonlyArray<number> | undefined;
   readonly nodes: ReadonlyArray<ASTNode> | undefined;
-  public originalError: Error | null | undefined;
+  public originalError: Error | undefined;
```

Comment:
To check my understanding: this is a safe change (that won't break users of graphql@15) because it's OK for an implementation of an interface to declare that a field has a type that is a subtype of the supertype's type?

Or I guess this is specifically the case because the interface (in v15) declared the field as `readonly`, which means that while ApolloError allows you to write `e.originalError = null`, that's unrelated to the GraphQLError declaration.  (Because `readonly` just means "this interface doesn't provide the ability for you to change the field", not "the implementing type must forbid you from changing the field".)

Right?

Reply:
@glasser Basically, we need to do this change to be compatible with v16.
From a purely theoretical position, it's a breaking change but I just can't figure out how to work around it.
In practical terms, I don't think it will break anyone especially since runtime behavior is the same.
Worst-case scenario library author (don't expect end-users to mess with `originalError`) need to change `e.originalError = null` to `e.originalError = undefined`.

Happy to replace it with some workaround that is less breaking I just can't come up with something right now.

Reply:
Ah right, I was focused just on "will this work with graphql 15" and not "is this a backwards incompatible change for users". But I think it's fine. We'll make this a minor update at least.